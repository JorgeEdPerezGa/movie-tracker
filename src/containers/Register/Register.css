.page-title {
  color: #959595;
  padding-top: 40px;
  font-weight: 200;
  margin: 0;
  font-size: 40px; }

.register {
  background: #1c1c1c;
  min-height: 100vh; }

.register-form {
  padding: 30px;
  padding-top: 150px;
  padding-top: 40px;
  display: flex;
  justify-content: center; }

.input-container {
  padding-top: 10px;
  display: flex;
  flex-direction: column;
  width: 500px;
  align-items: center;
  background: #c7c7c7; }

.register-input {
  font-size: 17px;
  margin-top: 34px;
  width: 400px;
  height: 30px;
  padding-left: 7px; }

.register-button {
  margin-bottom: 35px;
  background: #00afff;
  border-radius: 100px;
  border: none;
  color: white;
  margin-top: 29px;
  height: 37px;
  width: 150px;
  font-size: 15px; }
  .register-button:hover {
    cursor: pointer;
    background: #e7eff7;
    color: #090e12; }
