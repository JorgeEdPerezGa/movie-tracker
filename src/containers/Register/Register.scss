.page-title {
  color: #959595;
  padding-top: 40px;
  font-weight: 200;
  margin: 0;
  font-size: 40px;
}

.register {
  background: #1c1c1c;
  min-height: 100vh;
}

.register-form {
  padding: 30px;
  padding-top: 150px;
  padding-top: 40px;
  display: flex;
  justify-content: center;
}

.input-container{
  border: solid 3px #e2368d;
  padding-top: 10px;
  display: flex;
  flex-direction: column;
  width: 500px;
  align-items: center;
  background: #c7c7c7;
  // background: linear-gradient(230deg, rgb(75, 207, 147), rgb(75, 121, 207), rgb(162, 75, 207)) 0% 0% / 300% 300%;
  //
  // -webkit-animation: AnimationName 30s ease infinite;
  // animation: AnimationName 30s ease infinite;
  //
  // @-webkit-keyframes AnimationName {
  //     0%{background-position:0% 50%}
  //     50%{background-position:100% 50%}
  //     100%{background-position:0% 50%}
  // }
  // @keyframes AnimationName {
  //     0%{background-position:0% 50%}
  //     50%{background-position:100% 50%}
  //     100%{background-position:0% 50%}
  // }
}

.register-input {
  font-size: 17px;
  margin-top: 34px;
  width: 400px;
  height: 30px;
  padding-left: 7px;
}

.register-button {
  margin-bottom: 35px;
  background: #00afff;
  border-radius: 100px;
  border: none;
  color: white;
  margin-top: 29px;
  height: 37px;
  width: 150px;
  font-size: 15px;
  &:hover {
    cursor: pointer;
    background: #e7eff7;
    color: #090e12;
  }
}
