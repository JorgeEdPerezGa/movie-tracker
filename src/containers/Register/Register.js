import React, { Component } from 'react';
import { addUser } from '../../actions';
import { connect } from 'react-redux';
import { registerUser } from '../../helper';
import './Register.css';

export class Register extends Component {
  constructor(){
    super();
    this.state = {
      name: '',
      email: '',
      password1: '',
      password2: '',
      error: false,
      alert: ''
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({ [name]: value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { password1, password2, email, name } = this.state;

    if ( password1 === password2 ) {
      const user = { name, email, password: password1 }
      this.handleRegisterUser(user)
    } else { 
      this.setAlert("Passwords don't match")
    }
  }

  handleRegisterUser = async (user) => {
    try{
      const newUser = await registerUser(user);
      this.handleLogin(newUser.id);  
    } catch(error) {
      this.setAlert('User already exists')
    }
  }

  setAlert = (alert) => {
    this.setState({error: true, alert, name: '', email: '', password1: '', password2: '' })
  }
  
  handleLogin = (id) => {
    const { name, email, password1 } = this.state
    this.props.addUser({name, email, password: password1, id});
    this.props.history.push('/');
  }

  renderAlert = () => {
    const { error, alert } = this.state;
    setTimeout(() => { this.setState({error: false}) }, 10000);
    return error && ( <p color="white">{alert}</p> );
  }

  render() {
    return (
      <div className='register'>
        <h1 className='page-title'>GET STARTED</h1>
        <form
          onSubmit={this.handleSubmit}
          className='register-form'>
          <div className='input-container'>
            <input
              className='register-input'
              placeholder='NAME'
              name='name'
              type='text'
              value={this.state.name}
              onChange = {this.handleChange}/>
            <input
              className='register-input'
              placeholder='EMAIL'
              name='email'
              type='text'
              value={this.state.email}
              onChange = {this.handleChange}/>
            <input
              className='register-input user-password-1'
              placeholder='PASSWORD'
              name='password1'
              type='text'
              value={this.state.password1}
              onChange = {this.handleChange}/>
            <input
              className='register-input user-password-2'
              placeholder='RETYPE PASSWORD'
              name='password2'
              type='text'
              value={this.state.password2}
              onChange = {this.handleChange}/>
            <button className='register-button'>ENTER</button>
          </div>
        </form>
          { this.renderAlert() }
      </div>
    );
  }
}

const mapStateToProps = (store) => ({
  user: store.user
});

const mapDispatchToProps = (dispatch) => ({
  addUser: (user) => dispatch(addUser(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(Register);
