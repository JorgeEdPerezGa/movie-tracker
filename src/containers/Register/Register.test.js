import React from 'react';
import { shallow } from 'enzyme';
import { Register } from './Register';
jest.mock('../../helper');

describe('Register', () => {

  let renderedComponent;
  let mockLoginUser;
  let mockHistory;
  let mockUser;
  let mockAddUser;

  beforeEach(() => {
    mockLoginUser = jest.fn();
    mockHistory = { push: jest.fn() };
    mockAddUser = jest.fn();
    window.fetch = jest.fn();
    mockUser = {
      email: "kindoffake@123.com",
      id: 3,
      name: "fred",
      password1: "pw321",
      password2: "pw321"
    };

    renderedComponent = shallow(
      <Register
        user={mockUser}
        loginUser={mockLoginUser}
        history={mockHistory}
        addUser={mockAddUser}
      />
    );
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should start with the expected default state', () => {
    const expected = { name: '', email: '', password1: '', password2: '', alert: '', error: false };

    expect(renderedComponent.state()).toEqual(expected);
  });

  it('should set change state on handleChange', () => {
    expect(renderedComponent.state()).toEqual({ name: '', email: '', password1: '', password2: '', alert: '', error: false });

    renderedComponent.instance().handleChange({ target: { name: 'email', value: 'aaa' } });
    expect(renderedComponent.state()).toEqual({ email: 'aaa', name:'', password1:'', password2:'', alert: '', error: false});

    renderedComponent.instance().handleChange({ target: { name: 'password1', value: 'bbb' } });
    expect(renderedComponent.state()).toEqual({ email: 'aaa', name:'', password1: 'bbb', password2:'', alert: '', error: false});

    renderedComponent.instance().handleChange({ target: { name: 'password2', value: 'ccc' } });
    expect(renderedComponent.state()).toEqual({ email: 'aaa', name:'', password1: 'bbb', password2: 'ccc', alert: '', error: false });
  });

  it('handleSubmit should call registerUser with the expected params', () => {
    const state = {email: 'fake', name: 'blob', password1: 'uncrackable', password2: "uncrackable"};
    const expected = {email: 'fake', name: 'blob', password: 'uncrackable'};
    const mockEvent = { preventDefault: jest.fn() };

    renderedComponent.setState(state);

    expect(window.fetch).not.toHaveBeenCalled();

    renderedComponent.instance().handleSubmit(mockEvent);

    expect(window.fetch).toHaveBeenCalledWith(expected);
  });

  it('handleSubmit should setAlert with the expected params if the passwords dont match', () => {
    renderedComponent.setState({password1: 'a', password2:'b'});
    const mockEvent = { preventDefault: jest.fn() };

    renderedComponent.instance().handleSubmit(mockEvent);

    expect(renderedComponent.state('error')).toEqual(true);
    expect(renderedComponent.state('alert')).toEqual("Passwords don't match!");
  });

  it('handleSubmit should call loginUser with the expected params', () => {
    const state = {email: 'one', name:'asdf', password1: 'two', password2:'two'};
    const expected = {email: 'one', name:'asdf', password: 'two'};
    const mockEvent = { preventDefault: jest.fn() };

    renderedComponent.setState(state);

    expect(window.fetch).not.toHaveBeenCalled();

    renderedComponent.instance().handleSubmit(mockEvent);

    expect(window.fetch).toHaveBeenCalledWith(expected);
  });

  it('handleLogin should call addUser', () => {
    const expected = {
      email: "kindoffake@123.com",
      id: 3,
      name: "fred",
      password: "pw321"
    };
    expect(mockLoginUser).not.toHaveBeenCalled();

    renderedComponent.setState(mockUser);
    renderedComponent.instance().handleLogin(3);

    expect(mockAddUser).toHaveBeenCalledWith(expected);
  });

  it('handleRegisterUser should call registerUser with the expected params', () => {
    const state = {email: 'fake', name: 'blob', password1: 'uncrackable', password2: "uncrackable"};
    const expected = {email: 'fake', name: 'blob', password: 'uncrackable'};
    const mockEvent = { preventDefault: jest.fn() };

    renderedComponent.setState(state);

    expect(window.fetch).not.toHaveBeenCalled();

    renderedComponent.instance().handleSubmit(mockEvent);

    expect(window.fetch).toHaveBeenCalledWith(expected);
  });

  it('setAlert should setState as expected', () => {
    expect(renderedComponent.state('error')).toEqual(false);
    expect(renderedComponent.state('alert')).toEqual('');

    renderedComponent.instance().setAlert('Alert Message');
    renderedComponent.update();

    expect(renderedComponent.state('error')).toEqual(true);
    expect(renderedComponent.state('alert')).toEqual('Alert Message');
  });

  it('renderAlert should match snapshot', () => {
    renderedComponent.instance().setAlert('Alert message');
    renderedComponent.update();

    expect(renderedComponent).toMatchSnapshot();
  });
});
