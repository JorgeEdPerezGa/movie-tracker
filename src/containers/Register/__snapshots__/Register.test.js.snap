// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renderAlert should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    addUser={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    loginUser={[Function]}
    user={
        Object {
            "email": "kindoffake@123.com",
            "id": 3,
            "name": "fred",
            "password1": "pw321",
            "password2": "pw321",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="page-title"
>
          GET STARTED
</h1>,
        <form
          className="register-form"
          onSubmit={[Function]}
>
          <div
                    className="input-container"
          >
                    <input
                              className="register-input"
                              name="name"
                              onChange={[Function]}
                              placeholder="NAME"
                              type="text"
                              value=""
                    />
                    <input
                              className="register-input"
                              name="email"
                              onChange={[Function]}
                              placeholder="EMAIL"
                              type="text"
                              value=""
                    />
                    <input
                              className="register-input user-password-1"
                              name="password1"
                              onChange={[Function]}
                              placeholder="PASSWORD"
                              type="text"
                              value=""
                    />
                    <input
                              className="register-input user-password-2"
                              name="password2"
                              onChange={[Function]}
                              placeholder="RETYPE PASSWORD"
                              type="text"
                              value=""
                    />
                    <button
                              className="register-button"
                              disabled={true}
                    >
                              ENTER
                    </button>
          </div>
</form>,
        <p
          className="error-message"
          color="white"
>
          Alert message
</p>,
      ],
      "className": "register",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "GET STARTED",
          "className": "page-title",
        },
        "ref": null,
        "rendered": "GET STARTED",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-container"
>
            <input
                        className="register-input"
                        name="name"
                        onChange={[Function]}
                        placeholder="NAME"
                        type="text"
                        value=""
            />
            <input
                        className="register-input"
                        name="email"
                        onChange={[Function]}
                        placeholder="EMAIL"
                        type="text"
                        value=""
            />
            <input
                        className="register-input user-password-1"
                        name="password1"
                        onChange={[Function]}
                        placeholder="PASSWORD"
                        type="text"
                        value=""
            />
            <input
                        className="register-input user-password-2"
                        name="password2"
                        onChange={[Function]}
                        placeholder="RETYPE PASSWORD"
                        type="text"
                        value=""
            />
            <button
                        className="register-button"
                        disabled={true}
            >
                        ENTER
            </button>
</div>,
          "className": "register-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="register-input"
                name="name"
                onChange={[Function]}
                placeholder="NAME"
                type="text"
                value=""
/>,
              <input
                className="register-input"
                name="email"
                onChange={[Function]}
                placeholder="EMAIL"
                type="text"
                value=""
/>,
              <input
                className="register-input user-password-1"
                name="password1"
                onChange={[Function]}
                placeholder="PASSWORD"
                type="text"
                value=""
/>,
              <input
                className="register-input user-password-2"
                name="password2"
                onChange={[Function]}
                placeholder="RETYPE PASSWORD"
                type="text"
                value=""
/>,
              <button
                className="register-button"
                disabled={true}
>
                ENTER
</button>,
            ],
            "className": "input-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input",
                "name": "name",
                "onChange": [Function],
                "placeholder": "NAME",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "EMAIL",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input user-password-1",
                "name": "password1",
                "onChange": [Function],
                "placeholder": "PASSWORD",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input user-password-2",
                "name": "password2",
                "onChange": [Function],
                "placeholder": "RETYPE PASSWORD",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ENTER",
                "className": "register-button",
                "disabled": true,
              },
              "ref": null,
              "rendered": "ENTER",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alert message",
          "className": "error-message",
          "color": "white",
        },
        "ref": null,
        "rendered": "Alert message",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-title"
>
            GET STARTED
</h1>,
          <form
            className="register-form"
            onSubmit={[Function]}
>
            <div
                        className="input-container"
            >
                        <input
                                    className="register-input"
                                    name="name"
                                    onChange={[Function]}
                                    placeholder="NAME"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="register-input"
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="EMAIL"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="register-input user-password-1"
                                    name="password1"
                                    onChange={[Function]}
                                    placeholder="PASSWORD"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="register-input user-password-2"
                                    name="password2"
                                    onChange={[Function]}
                                    placeholder="RETYPE PASSWORD"
                                    type="text"
                                    value=""
                        />
                        <button
                                    className="register-button"
                                    disabled={true}
                        >
                                    ENTER
                        </button>
            </div>
</form>,
          <p
            className="error-message"
            color="white"
>
            Alert message
</p>,
        ],
        "className": "register",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GET STARTED",
            "className": "page-title",
          },
          "ref": null,
          "rendered": "GET STARTED",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-container"
>
              <input
                            className="register-input"
                            name="name"
                            onChange={[Function]}
                            placeholder="NAME"
                            type="text"
                            value=""
              />
              <input
                            className="register-input"
                            name="email"
                            onChange={[Function]}
                            placeholder="EMAIL"
                            type="text"
                            value=""
              />
              <input
                            className="register-input user-password-1"
                            name="password1"
                            onChange={[Function]}
                            placeholder="PASSWORD"
                            type="text"
                            value=""
              />
              <input
                            className="register-input user-password-2"
                            name="password2"
                            onChange={[Function]}
                            placeholder="RETYPE PASSWORD"
                            type="text"
                            value=""
              />
              <button
                            className="register-button"
                            disabled={true}
              >
                            ENTER
              </button>
</div>,
            "className": "register-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="register-input"
                  name="name"
                  onChange={[Function]}
                  placeholder="NAME"
                  type="text"
                  value=""
/>,
                <input
                  className="register-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="EMAIL"
                  type="text"
                  value=""
/>,
                <input
                  className="register-input user-password-1"
                  name="password1"
                  onChange={[Function]}
                  placeholder="PASSWORD"
                  type="text"
                  value=""
/>,
                <input
                  className="register-input user-password-2"
                  name="password2"
                  onChange={[Function]}
                  placeholder="RETYPE PASSWORD"
                  type="text"
                  value=""
/>,
                <button
                  className="register-button"
                  disabled={true}
>
                  ENTER
</button>,
              ],
              "className": "input-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input",
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "NAME",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "EMAIL",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input user-password-1",
                  "name": "password1",
                  "onChange": [Function],
                  "placeholder": "PASSWORD",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input user-password-2",
                  "name": "password2",
                  "onChange": [Function],
                  "placeholder": "RETYPE PASSWORD",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ENTER",
                  "className": "register-button",
                  "disabled": true,
                },
                "ref": null,
                "rendered": "ENTER",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alert message",
            "className": "error-message",
            "color": "white",
          },
          "ref": null,
          "rendered": "Alert message",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Register should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    addUser={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    loginUser={[Function]}
    user={
        Object {
            "email": "kindoffake@123.com",
            "id": 3,
            "name": "fred",
            "password1": "pw321",
            "password2": "pw321",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="page-title"
>
          GET STARTED
</h1>,
        <form
          className="register-form"
          onSubmit={[Function]}
>
          <div
                    className="input-container"
          >
                    <input
                              className="register-input"
                              name="name"
                              onChange={[Function]}
                              placeholder="NAME"
                              type="text"
                              value=""
                    />
                    <input
                              className="register-input"
                              name="email"
                              onChange={[Function]}
                              placeholder="EMAIL"
                              type="text"
                              value=""
                    />
                    <input
                              className="register-input user-password-1"
                              name="password1"
                              onChange={[Function]}
                              placeholder="PASSWORD"
                              type="text"
                              value=""
                    />
                    <input
                              className="register-input user-password-2"
                              name="password2"
                              onChange={[Function]}
                              placeholder="RETYPE PASSWORD"
                              type="text"
                              value=""
                    />
                    <button
                              className="register-button"
                              disabled={true}
                    >
                              ENTER
                    </button>
          </div>
</form>,
        false,
      ],
      "className": "register",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "GET STARTED",
          "className": "page-title",
        },
        "ref": null,
        "rendered": "GET STARTED",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-container"
>
            <input
                        className="register-input"
                        name="name"
                        onChange={[Function]}
                        placeholder="NAME"
                        type="text"
                        value=""
            />
            <input
                        className="register-input"
                        name="email"
                        onChange={[Function]}
                        placeholder="EMAIL"
                        type="text"
                        value=""
            />
            <input
                        className="register-input user-password-1"
                        name="password1"
                        onChange={[Function]}
                        placeholder="PASSWORD"
                        type="text"
                        value=""
            />
            <input
                        className="register-input user-password-2"
                        name="password2"
                        onChange={[Function]}
                        placeholder="RETYPE PASSWORD"
                        type="text"
                        value=""
            />
            <button
                        className="register-button"
                        disabled={true}
            >
                        ENTER
            </button>
</div>,
          "className": "register-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="register-input"
                name="name"
                onChange={[Function]}
                placeholder="NAME"
                type="text"
                value=""
/>,
              <input
                className="register-input"
                name="email"
                onChange={[Function]}
                placeholder="EMAIL"
                type="text"
                value=""
/>,
              <input
                className="register-input user-password-1"
                name="password1"
                onChange={[Function]}
                placeholder="PASSWORD"
                type="text"
                value=""
/>,
              <input
                className="register-input user-password-2"
                name="password2"
                onChange={[Function]}
                placeholder="RETYPE PASSWORD"
                type="text"
                value=""
/>,
              <button
                className="register-button"
                disabled={true}
>
                ENTER
</button>,
            ],
            "className": "input-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input",
                "name": "name",
                "onChange": [Function],
                "placeholder": "NAME",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "EMAIL",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input user-password-1",
                "name": "password1",
                "onChange": [Function],
                "placeholder": "PASSWORD",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-input user-password-2",
                "name": "password2",
                "onChange": [Function],
                "placeholder": "RETYPE PASSWORD",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ENTER",
                "className": "register-button",
                "disabled": true,
              },
              "ref": null,
              "rendered": "ENTER",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-title"
>
            GET STARTED
</h1>,
          <form
            className="register-form"
            onSubmit={[Function]}
>
            <div
                        className="input-container"
            >
                        <input
                                    className="register-input"
                                    name="name"
                                    onChange={[Function]}
                                    placeholder="NAME"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="register-input"
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="EMAIL"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="register-input user-password-1"
                                    name="password1"
                                    onChange={[Function]}
                                    placeholder="PASSWORD"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="register-input user-password-2"
                                    name="password2"
                                    onChange={[Function]}
                                    placeholder="RETYPE PASSWORD"
                                    type="text"
                                    value=""
                        />
                        <button
                                    className="register-button"
                                    disabled={true}
                        >
                                    ENTER
                        </button>
            </div>
</form>,
          false,
        ],
        "className": "register",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GET STARTED",
            "className": "page-title",
          },
          "ref": null,
          "rendered": "GET STARTED",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-container"
>
              <input
                            className="register-input"
                            name="name"
                            onChange={[Function]}
                            placeholder="NAME"
                            type="text"
                            value=""
              />
              <input
                            className="register-input"
                            name="email"
                            onChange={[Function]}
                            placeholder="EMAIL"
                            type="text"
                            value=""
              />
              <input
                            className="register-input user-password-1"
                            name="password1"
                            onChange={[Function]}
                            placeholder="PASSWORD"
                            type="text"
                            value=""
              />
              <input
                            className="register-input user-password-2"
                            name="password2"
                            onChange={[Function]}
                            placeholder="RETYPE PASSWORD"
                            type="text"
                            value=""
              />
              <button
                            className="register-button"
                            disabled={true}
              >
                            ENTER
              </button>
</div>,
            "className": "register-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="register-input"
                  name="name"
                  onChange={[Function]}
                  placeholder="NAME"
                  type="text"
                  value=""
/>,
                <input
                  className="register-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="EMAIL"
                  type="text"
                  value=""
/>,
                <input
                  className="register-input user-password-1"
                  name="password1"
                  onChange={[Function]}
                  placeholder="PASSWORD"
                  type="text"
                  value=""
/>,
                <input
                  className="register-input user-password-2"
                  name="password2"
                  onChange={[Function]}
                  placeholder="RETYPE PASSWORD"
                  type="text"
                  value=""
/>,
                <button
                  className="register-button"
                  disabled={true}
>
                  ENTER
</button>,
              ],
              "className": "input-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input",
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "NAME",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "EMAIL",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input user-password-1",
                  "name": "password1",
                  "onChange": [Function],
                  "placeholder": "PASSWORD",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "register-input user-password-2",
                  "name": "password2",
                  "onChange": [Function],
                  "placeholder": "RETYPE PASSWORD",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ENTER",
                  "className": "register-button",
                  "disabled": true,
                },
                "ref": null,
                "rendered": "ENTER",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
